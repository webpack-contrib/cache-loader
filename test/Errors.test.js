import loader, { pitch } from '../src';

// Needed for `schema-utils` to not call
// process.exit(1) when running tests
process.env.JEST = true;

describe('Errors', () => {
  test('Validation Error', () => {
    const err = () => loader.call({ query: { cacheIdentifier: 1 } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('Validation Error (Pitch)', () => {
    const err = () => pitch.call({ query: { cacheIdentifier: 1 } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });
});
