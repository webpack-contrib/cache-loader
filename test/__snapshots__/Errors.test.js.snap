// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Validation Error (Pitch) 1`] = `
"Cache Loader (Pitch) Invalid Options

options.cacheIdentifier should be string
"
`;

exports[`Errors Validation Error 1`] = `
"Cache Loader Invalid Options

options.cacheIdentifier should be string
"
`;
